firebase-vscode/src/analytics.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  env,
+  TelemetryLogger,
+  TelemetrySender,
+} from "vscode";
+import { pluginLogger } from "./logger-wrapper";
+import { AnalyticsParams, trackVSCode } from "./track";
+
+
+export class AnalyticsLogger {
+    readonly logger: TelemetryLogger
+    constructor() {
+        this.logger = env.createTelemetryLogger(new GA4TelemetrySender(pluginLogger));
+    }
+}
+
+class GA4TelemetrySender implements TelemetrySender {
+  constructor(readonly pluginLogger) {}
+
+  sendEventData(
+    eventName: string,
+    data?: Record<string, any> | undefined,
+  ): void {
+    if (!env.isTelemetryEnabled) {
```

Comment:
Quick sanity check - Is this a VSCode wide setting? If so, users need to explicitly opt in for isTelemetryEnabled to be true, right?

Reply:
correct. I need to look into how to prompt users to enable. But this is a must check for metrics within vscode